### PR TITLE
refactor(frontend): use existing balance prop to filter tokens

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

Instead of calling again the balances store, we can use directly the `usdBalance` prop of type `TokenUi`.

# Tests

Local replica unchanged:

All balances

Screenshot 2024-08-28 at 07 36 02

Only non-zero balances
Screenshot 2024-08-28 at 07 36 10

### CODE REVIEW

File: src/frontend/src/lib/components/tokens/TokensDisplayHandler.svelte [hunk 1]
```diff
@@ -16,8 +14,7 @@
 
 	let sortedTokens: TokenUi[];
 	$: sortedTokens = $combinedDerivedSortedNetworkTokensUi.filter(
-		({ id: tokenId }) =>
-			($balancesStore?.[tokenId]?.data ?? BigNumber.from(0n)).gt(0n) || displayZeroBalance
+		({ usdBalance }) => (usdBalance ?? 0) || displayZeroBalance
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/tokens/TokensDisplayHandler.svelte:17]: It's not the same logic anymore - i.e. there is no comparison that the balance is greater than zero - is that expected?

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/tokens/TokensDisplayHandler.svelte:17]: The value `0` is typically a `falsy` value, meaning that we can use it as direct condition (and the balance will never be negative, but even if it is, we should probably show it either way)

in any case, to have the scope more clear in the code, i can put the `>0` condition, if you prefer

Comment (peterpeterparker) [reply src/frontend/src/lib/components/tokens/TokensDisplayHandler.svelte:17]: If the behavior remains intact and the change is excepted for sure it's fine.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

**Pull request merged**